modules/angular2/src/core/compiler/view.ts

```
@@ -202,7 +209,35 @@ export class AppView implements ChangeDispatcher, RenderEventDispatcher {
 
   getHostElement(): ElementRef {
     var boundElementIndex = this.mainMergeMapping.hostElementIndicesByViewIndex[this.viewOffset];
-    return this.elementRefs[boundElementIndex];
+    return isPresent(boundElementIndex) ? this.elementRefs[boundElementIndex] : null;
+  }
+
+  getDebugContext(elementIndex: number, directiveIndex: DirectiveIndex): StringMap<string, any> {
+    try {
+      var offsettedIndex = this.elementOffset + elementIndex;
+      var hasRefForIndex = offsettedIndex < this.elementRefs.length;
+
+      var elementRef = hasRefForIndex ? this.elementRefs[this.elementOffset + elementIndex] : null;
+      var host = this.getHostElement();
+      var ei = hasRefForIndex ? this.elementInjectors[this.elementOffset + elementIndex] : null;
+
+      var element = isPresent(elementRef) ? elementRef.nativeElement : null;
+      var componentElement = isPresent(host) ? host.nativeElement : null;
+      var directive = isPresent(directiveIndex) ? this.getDirectiveFor(directiveIndex) : null;
+      var injector = isPresent(ei) ? ei.getInjector() : null;
+
+      return {
+        element: element,
+        componentElement: componentElement,
+        directive: directive,
+        context: this.context,
+        locals: _localsToStringMap(this.locals),
+        injector: injector
+      };
+    } catch (e) {
+      // if an error happens during getting the debug context, we return an empty map.
+      return {};
```

Or perhaps an special error context instance? What kind of debugging tool is it if it swallows its own errors? :)